.github/workflows/sel4test-sim.yml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021, Proofcraft Pty Ltd
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# sel4test simulation runs
+#
+# See sel4test-sim/builds.yml in the repo seL4/ci-actions for configs.
+
+name: seL4Test-Sim
+
+on:
+  workflow_call:
+
+# To reduce the load we cancel any older runs of this workflow for the current
+# PR. Ok on the workflow level here, because this workflow does not re-trigger
+# on "labeled" events.
+concurrency:
+  group: ${{ github.workflow }}-sel4sim-pr-${{ github.event.number }}
```

Comment:
Debugging gives me `seL4Test-sel4sim-pr-` here when a never job cancels an old run. Debugging says
```
  "event_name": "pull_request",
    "event": {
      "action": "synchronize",
      ...
      "number": 140,      
      ...
```
so I'm a but puzzled why the PR number is missing there.

Reply:
Seems the whole `github.event` does not exist a this stage.

Reply:
In which scenario is that happening? I.e. are you seeing this in the old job that is being cancelled or in the new job that is the reason for cancellation?

Reply:
I'm seeing this in the old job, where github gives the cancellation reason. This is the only place where the group name is ever printed. It seems impossible to see the concurrency group for a running job anywhere. You can print it in a job/step, but the context there seems slightly different, because this runs on the runner. The workflow control and concurrency seems to run on an "action server", which does not see some dynamic parts of the contexts. The github documentation is quite thin here, but that is what I found in some github community postings.

Reply:
Ok, the docs do say that `github`, `inputs`, and `vars` are available at the [workflow level](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency), and additionally `needs`, `strategy`, and `matrix` on the [job level](https://docs.github.com/en/actions/using-jobs/using-concurrency#using-concurrency-in-different-scenarios). So I think we are fine with this expression, it's just hard to debug as usual on GitHub.

Reply:
Let's see if I get some answers here: https://github.com/orgs/community/discussions/107552

Reply:
I'm going to put it into the job level for now, so that we can merge this. Would still be good to figure out what exactly is available, though.

Reply:
The other option could be adding another callable workflow layer here. The the job level concurrency group there will cover everything. It could also filter the events and invoke this workflow with a simple bool parameter for enabling hw-test.